driver-core/src/main/com/mongodb/client/model/Aggregates.java

```diff
@@ -70,24 +70,105 @@ public static Bson addFields(final List<Field<?>> fields) {
+     * @param mergeFunction        a function used to merge two internal states, e.g. accumulated on different shards or threads. It
+     *                             returns the resulting state of the accumulator.
+     * @param finalizeFunction     a function used to finalize the state and return the result (may be null)
+     * @param lang                 a language specifier
+     * @return the $accumulator pipeline stage
+     * @mongodb.driver.manual reference/operator/aggregation/accumulator/ $accumulator
+     * @mongodb.server.release 4.4
+     * @since 4.1
+     */
+    public static Bson accumulator(final String initFunction, final String accumulateFunction, final String mergeFunction,
+                                   @Nullable final String finalizeFunction, final String lang) {
+        return accumulator(initFunction, null, accumulateFunction, null, mergeFunction, finalizeFunction, lang);
+    }
+
+    /**
+     * Creates an $accumulator pipeline stage
+     *
+     * @param initFunction         a function used to initialize the state
+     * @param initArgs             init function’s arguments (may be null)
+     * @param accumulateFunction   a function used to accumulate documents
+     * @param accumulateArgs       additional accumulate function’s arguments (may be null). The first argument to the function
+     *                             is ‘state’.
+     * @param mergeFunction        a function used to merge two internal states, e.g. accumulated on different shards or threads. It
+     *                             returns the resulting state of the accumulator.
+     * @param finalizeFunction     a function used to finalize the state and return the result (may be null)
      * @param lang                 a language specifier
      * @return the $accumulator pipeline stage
      * @mongodb.driver.manual reference/operator/aggregation/accumulator/ $accumulator
      * @mongodb.server.release 4.4
      * @since 4.1
      */
-    public static Bson accumulator(final Field<?> initField, @Nullable final Field<?> initArgsField, final Field<?> accumulateField,
-                                   @Nullable final Field<?> accumulateArgsField, final Field<?> mergeField,
-                                   @Nullable final Field<?> finalizeField, final Field<?> lang) {
-        return new FieldsStage("$accumulator", asList(initField, initArgsField, accumulateField, accumulateArgsField,
-                mergeField, finalizeField, lang));
+    public static Bson accumulator(final String initFunction, @Nullable final List<String> initArgs, final String accumulateFunction,
+                                   @Nullable final List<String> accumulateArgs, final String mergeFunction,
+                                   @Nullable final String finalizeFunction, final String lang) {
+        return new FieldsStage("$accumulator", asList(new Field<String>("init", initFunction),
```

Comment:
Do we need FieldsStage here or could this just return a Document representing the accumulator?

Reply:
You're right, of course. I have converted the return value to a BsonDocument.